kong/db/declarative/init.lua

```diff
@@ -663,16 +660,15 @@ function declarative.load_into_cache(entities, meta, hash, shadow)
         end
       end
 
-      local ok, err = kong.core_cache:safe_set(cache_key, item, shadow)
-      if not ok then
+      local item_marshalled, err = marshall(item)
+      if not item_marshalled then
         return nil, err
       end
 
+      t:set(cache_key, item_marshalled)
```

Comment:
The `:set` is a low level LMDB interface, and it will only accept string input for performance reasons (not all use cases needs marshaling, after all). So in this case Kong will perform the marshaling itself before handing over to LMDB (with a small perf benefit). But perf is not the reason for doing the marshaling separately.